### PR TITLE
ZPI-35-create-sign-in-and-sign-up-page

### PR DESCRIPTION
**Pull request by Maciag312**
- add authorization api client
- add allow page

### CODE REVIEW

File: src/App.tsx [hunk 1]
```diff
@@ -1,23 +1,30 @@
...
-          href="https://reactjs.org"
-          target="_blank"
-          rel="noopener noreferrer"
-        >
-          Learn React
-        </a>
+        <ChakraProvider>
+          <Router>
+            <Switch>
+              <Route exact path="/auth">
```
Comment (bachm44) [inline src/App.tsx:16]: Move route paths to one file called routes.ts

File: src/App.tsx [hunk 2]
```diff
@@ -1,23 +1,30 @@
...
 import './App.css';
+import AllowPage from './user/authorization/AllowPage';
+import SignUpPage from './user/authorization/SignUpPage';
+import SignInPage from './user/authorization/SignInPage';
+import { ChakraProvider } from "@chakra-ui/react"
+import {Switch, BrowserRouter as Router, Route} from 'react-router-dom'
 
 function App() {
   return (
     <div className="App">
```
Comment (bachm44) [inline src/App.tsx:11]: Use `<Box>` or `<div>` but not both

**Review by bachm44 — CHANGES_REQUESTED** [review]

File: src/user/authorization/SignInPage.tsx [hunk 3]
```diff
@@ -0,0 +1,87 @@
...
+                                <Input
+                                    pr="4.5rem"
+                                    type={show ? "text" : "password"}
+                                    bgColor="white"
+                                    onChange={event => setPassword(event.currentTarget.value)}
+                                />
+                        
+                                <InputRightElement width="4.5rem">
+                                    <Button h="1.75rem" size="sm" onClick={handleClick}>
+                                    {show ? "Hide" : "Show"}
```
Comment (bachm44) [inline src/user/authorization/SignInPage.tsx:76]: Consider using internationalization library (ex. i18n) to keep all strings in one place

File: src/user/authorization/SignUpPage.tsx [hunk 4]
```diff
@@ -0,0 +1,86 @@
...
+    }
+  
+    return (
+        <>
+            <Box className="AuthorizationPageBox" rounded="lg">
+                <Box className="AuthorizationPageBoxContent">
+                    <Heading as="h3" size="lg" paddingTop="20px" marginBottom="5px">Sign up</Heading>
+                    <Text fontSize="2xl" marginBottom="25px" className="Logo">service logo</Text>
+                        <FormControl isRequired mt={6}>
+                            <FormLabel textAlign="left" mb="8px"> Username or email</FormLabel>
```
Comment (bachm44) [inline src/user/authorization/SignUpPage.tsx:53]: There is a space at the front of text. Consider i18n
```suggestion
                            <FormLabel textAlign="left" mb="8px">Username or email</FormLabel>
```

File: src/user/authorization/SignUpPage.tsx [hunk 5]
```diff
@@ -0,0 +1,86 @@
...
+                                />
+                        
+                                <InputRightElement width="4.5rem">
+                                    <Button h="1.75rem" size="sm" onClick={handleClick}>
+                                    {show ? "Hide" : "Show"}
+                                    </Button>
+                                </InputRightElement>
+                            </InputGroup>
+                        </FormControl>
+                        <Button  onClick={handleSubmit} colorScheme="green" marginTop="20px" size="sm">Sign up</Button>
```
Comment (bachm44) [inline src/user/authorization/SignUpPage.tsx:79]: Prettier or other formatting would be nice

```suggestion
                        <Button onClick={handleSubmit} colorScheme="green" marginTop="20px" size="sm">Sign up</Button>
```

**Review by bachm44 — CHANGES_REQUESTED** [review]

File: .gitignore [hunk 6]
```diff
@@ -21,3 +21,6 @@
 npm-debug.log*
 yarn-debug.log*
 yarn-error.log*
+
+package-lock.json
+yarn.lock
```
Comment (bachm44) [inline .gitignore:26]: This should not be in .gitignore

**Review by bachm44 — CHANGES_REQUESTED** [review]

File: src/user/authorization/SignUpPage.tsx [hunk 7]
```diff
@@ -0,0 +1,86 @@
...
+    const handleClick = () => setShow(!show)
+  
+    const [username, setUsername] = React.useState("")
+    const [password, setPassword] = React.useState("")
+
+    const handleSubmit = () => {
+        let creds: Creds
+        creds = {username: username, password: password}
+    
+        authorizationClientInstance.signUp(creds).then(
```
[the remark this answers is not in the document]
Comment (Maciag312) [reply src/user/authorization/SignUpPage.tsx:24]: toast cannot be inside class, where logic functions belongs, let's discuss it

File: src/user/authorization/api/AuthorizationClient.tsx [hunk 8]
```diff
@@ -0,0 +1,9 @@
+import { AxiosResponse } from "axios";
+import Creds from "../Creds";
+
+export interface AuthorizationClient {
+    signIn(creds: Creds): Promise<AxiosResponse<any>>
```
[the remark this answers is not in the document]
Comment (Maciag312) [reply src/user/authorization/api/AuthorizationClient.tsx:5]: Contract between client and server is not set up yet

**Pull request merged**